src/components/calcite-avatar/calcite-avatar.tsx

```diff
@@ -0,0 +1,129 @@
+import { Component, Element, h, Host, Prop, State } from "@stencil/core";
+import { getElementDir } from "../../utils/dom";
+import { hexToRGB, isValidHex } from "../calcite-color/utils";
+import { stringToHex } from "./utils";
+
+@Component({
+  tag: "calcite-avatar",
+  styleUrl: "calcite-avatar.scss",
+  shadow: true
+})
+export class CalciteAvatar {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLCalciteAvatarElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /** Select theme (light or dark) */
+  @Prop({ reflect: true }) theme: "light" | "dark" = "light";
+
+  /** specify the scale of the avatar, defaults to m */
+  @Prop({ mutable: true, reflect: true }) scale: "s" | "m" | "l" = "m";
+
+  /** src to an image (remember to add a token if the user is private) */
+  @Prop() thumbnail: string;
+
+  /** full name of the user */
+  @Prop() fullName: string;
+
+  /** user name */
+  @Prop() username: string;
+
+  /** unique id for user */
+  @Prop() userId: string;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  connectedCallback() {
+    // prop validations
+    const scale = ["s", "m", "l"];
+    if (!scale.includes(this.scale)) this.scale = "m";
+
+    const theme = ["dark", "light"];
+    if (!theme.includes(this.theme)) this.theme = "light";
+  }
+
+  render() {
+    const dir = getElementDir(this.el);
+    const content = this.determineContent();
+    return <Host dir={dir}>{content}</Host>;
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private State/Props
+  //
+  //--------------------------------------------------------------------------
+
+  /** True if thumnail fails to load */
+  @State() error = false;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+
+  private determineContent() {
+    if (this.thumbnail && !this.error) {
+      return <img class="thumbnail" onError={() => (this.error = true)} src={this.thumbnail} />;
```

Comment:
Does this need an `alt` tag? Should there be some A11y tests setup?

Reply:
This is a good point. I think the correct move is an empty `alt` tag as this will generally be supplemental to the name in writing somewhere? 

IDK, I guess it could be used both ways. We could make it configurable?

Reply:
> Should there be some A11y tests setup?

Does this refer to `e2e` a11y test?

Reply:
Yep 👍

Reply:
added a test with thumbnail and an empty `alt`. I think empty alt to mark as presentational is probably the move that makes the most sense right now as most of the time the avatar will appear right next to the name. (also added an aria-hidden on the initials). 

If dev's make the avatar interactive or give it a tooltip or hover text they will most likely be labeling the element at that point anyways ie with an `aria-label`